service/controller/resource/tcnp/template/params_main_auto_scaling_group.go

```diff
@@ -1,16 +1,25 @@
 package template
 
 type ParamsMainAutoScalingGroup struct {
-	AvailabilityZones     []string
-	Cluster               ParamsMainAutoScalingGroupCluster
-	DesiredCapacity       int
-	MaxBatchSize          string
-	MaxSize               int
-	MinInstancesInService string
-	MinSize               int
-	Subnets               []string
+	AvailabilityZones                   []string
+	Cluster                             ParamsMainAutoScalingGroupCluster
+	DesiredCapacity                     int
+	MaxBatchSize                        string
+	MaxSize                             int
+	MinInstancesInService               string
+	MinSize                             int
+	Subnets                             []string
+	OnDemandPercentageAboveBaseCapacity int
+	OnDemandBaseCapacity                int
+	SpotAllocationStrategy              string
+	LaunchTemplateOverrides             []LaunchTemplateOverride
 }
 
 type ParamsMainAutoScalingGroupCluster struct {
 	ID string
 }
+
+type LaunchTemplateOverride struct {
+	InstanceType     string
+	WeightedCapacity string
```

Comment:
Please add some comments as to what this is - `WeightedCapacity` as a string currently gives me no expectation what should be here.

Reply:
I've added quite a few comments for the new fields